alchemy/alchemy.py

```diff
@@ -884,24 +947,32 @@ def _alchemicallyModifyNonbondedForce(self, system, reference_force):
             c_rf = r_cutoff**(-1) * ((3*epsilon_solvent) / (2*epsilon_solvent + 1))
             electrostatics_energy_expression += "k_rf = %f;" % (k_rf.value_in_unit_system(unit.md_unit_system))
             electrostatics_energy_expression += "c_rf = %f;" % (c_rf.value_in_unit_system(unit.md_unit_system))
-        elif method in [openmm.NonbondedForce.PME, openmm.NonbondedForce.Ewald]:
+        elif nonbonded_method in [openmm.NonbondedForce.PME, openmm.NonbondedForce.Ewald]:
             # Ewald direct-space electrostatics
             [alpha_ewald, nx, ny, nz] = reference_force.getPMEParameters()
             if alpha_ewald == 0.0:
                 # If alpha is 0.0, alpha_ewald is computed by OpenMM from from the error tolerance.
                 [alpha_ewald, nx, ny, nz] = reference_force.getPMEParameters()
+                assert (alpha_ewald != 0.0) # we can't have alpha be zero
```

Comment:
Not sure why that assert is there (@jchodera?), but it looks like we can move the `assert` outside the if-clause?